include/util/exception.hpp

```diff
@@ -40,8 +41,14 @@ namespace util
 class exception final : public std::exception
 {
   public:
-    explicit exception(const char *message) : message(message) {}
-    explicit exception(std::string message) : message(std::move(message)) {}
+    explicit exception(const char *message_) : message(message_)
+    {
+        util::SimpleLogger().Write(logERROR) << message;
```

Comment:
I am not sure here if this is such a good idea. @daniel-j-h, whats your take? The logger is a synchronised entity and from what I can tell not exception save. So throwing an exception while throwing an exception looks like opening the doors of hell to me.